src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/service/ReferralService.kt

```diff
@@ -26,7 +27,7 @@ class ReferralService(val repository: ReferralRepository) {
 
     update.serviceCategoryId?.let {
       if (referral.serviceCategoryID != null) {
-        errors.add(FieldError(field = "serviceCategoryId", error = Code.SERVICE_CATEGORY_CANNOT_BE_CHANGED))
+        errors.add(FieldError(field = "serviceCategoryId", error = Code.FIELD_CANNOT_BE_CHANGED))
```

Comment:
This sounds like a business rule. Can we add -- for the API client and posterity's benefit -- the reason for having this rule, please? Perhaps as a message/reason?

Could this be `referral.serviceCategoryID != null  && referral.serviceCategoryID != update.serviceCategoryId` to allow idempotent updates from the client side?

Reply:
the reason for this is that things like outcomes and complexity level are based on the service category. so changing the service category could lead to inconsistent data in the referral, e.g. complexity levels for the wrong service category.

on the second part - yes, we can do that.

Reply:
Thank you, that's really useful context.

Would similar messaging make sense?
```kt
errors.add(FieldError(field = "serviceCategoryId", error = Code.FIELD_CANNOT_BE_CHANGED,
  reason = "service category can only be selected at the start of a new referral"))
```

The intent of this would be to be explicit to code readers and API clients about _why_ the field is immutable. (Could be done via data structures/predicate functions too)